nibabel/parrec.py

```diff
@@ -725,19 +734,26 @@ def get_bvals_bvecs(self):
         n_slices, n_vols = self.get_data_shape()[-2:]
         bvals = self.image_defs['diffusion_b_factor'][reorder].reshape(
             (n_slices, n_vols), order='F')
-        # All bvals within volume should be the same
-        assert not np.any(np.diff(bvals, axis=0))
+        if not self.strict_sort:
```

Comment:
The check should also be performed in the strict sort case since these lines occur after the sort, so this change was a mistake!  I think these checks should then stay in this function.